src/common/SurgeError.h

```diff
@@ -5,10 +5,13 @@
 class SurgeError {
 public:
    SurgeError(const std::string &message);
+   SurgeError(const std::string &title, const std::string &msg);
```

Comment:
You have an abbreviation `msg` here. Please just write it as `message` (same name as the member variable has). Also, the parameters are in wrong order.

I would define here:

```
static const std::string DEFAULT_TITLE ;

SurgeError(const std:;string& message, const std::string &title = DEFAULT_TITLE);
```
And declare in the CPP-file:

```
static const std::string SurgeError::DEFAULT_TITLE = "An error has occured";
```
Having two constructors for such a simple class is an overkill.

Reply:
Agree. And will do so